hpolib/container/client_abstract_benchmark.py

```diff
@@ -208,61 +214,109 @@ def download_container(container_dir, container_name, container_source):
+            fidelity = fidelity.get_dictionary()
+        elif isinstance(fidelity, dict):
+            fidelity = fidelity
+        else:
+            raise ValueError(f'Type of fidelity not understood: {type(fidelity)}')
+        f_str = json.dumps(fidelity, indent=None)
+        return f_str
+
     def objective_function(self, configuration: Union[np.ndarray, List, CS.Configuration, Dict],
+                           fidelity: Union[CS.Configuration, Dict, None] = None,
                            rng: Union[np.random.RandomState, int, None] = None, **kwargs) -> Dict:
-        """ Run a configuration for a given budget on the benchmark.
-        The parameter name for the budget may vary from benchmark to benchmark. Please use the parameter name
-        specified in the benchmark.
         """
+        Run a given configuration for a given fidelity on the containerized benchmark.
 
-        if rng is not None:
-            rng = self._cast_random_state_to_int(rng)
-            kwargs['rng'] = rng
+        Convert the given parameters to strings and send them via Pyro to the container.
+        Read the result information and parse them.
+
+        Parameters
+        ----------
+        configuration : np.ndarray, List, CS.Configuration, Dict
+        fidelity : CS.Configuration, Dict, None
+        rng : np.random.RandomState, int, None
+        kwargs : Dict
+
+        Returns
+        -------
+        Dict
+        """
+
+        kwargs_str = self._parse_kwargs(rng, **kwargs)
+        f_str = self._parse_fidelities(fidelity)
 
         if isinstance(configuration, np.ndarray):
             configuration = configuration.tolist()
 
         if isinstance(configuration, list):
-            configuration_str = json.dumps(configuration, indent=None)
-            json_str = self.benchmark.objective_function_list(configuration_str, json.dumps(kwargs))
+            c_str = json.dumps(configuration, indent=None)
+            json_str = self.benchmark.objective_function_list(c_str, f_str, kwargs_str)
             return json.loads(json_str)
         elif isinstance(configuration, CS.Configuration):
             c_str = json.dumps(configuration.get_dictionary(), indent=None)
-            json_str = self.benchmark.objective_function(c_str, json.dumps(kwargs))
-            return json.loads(json_str)
         elif isinstance(configuration, dict):
             c_str = json.dumps(configuration, indent=None)
-            json_str = self.benchmark.objective_function(c_str, json.dumps(kwargs))
-            return json.loads(json_str)
         else:
             raise ValueError(f'Type of config not understood: {type(configuration)}')
 
+        json_str = self.benchmark.objective_function(c_str, f_str, kwargs_str)
+        return json.loads(json_str)
+
     def objective_function_test(self, configuration: Union[np.ndarray, List, CS.Configuration, Dict],
+                                fidelity: Union[CS.Configuration, Dict, None] = None,
```

Comment:
I don't care whether we accept fidelity as ConfigurationSpace object, but would this work fine since in the other interfaces we only allow dict,None?

Reply:
Yes, this works fine, because in the wrapper function I catch this case and convert the configuration then to a dictionary.

Reply:
I thought it might be user friendly if we support also configurations. Then the user does not have to think about why ConfigSpace.Configurations are allowed as configuration and why not for the fidelities.